packages/fiori-elements-writer/src/index.ts

```diff
@@ -42,18 +44,36 @@ async function generate<T extends {}>(basePath: string, data: FioriElementsApp<T
 
     await addOdataService(basePath, feApp.service, fs);
 
+    const coercedUI5Version = semVer.coerce(feApp.ui5?.version)!;
     const templateOptions: TemplateOptions = {
-        changesPreview: feApp.ui5?.version
-            ? semVer.lt(semVer.coerce(feApp.ui5?.version)!, changesPreviewToVersion)
-            : false,
+        changesPreview: feApp.ui5?.version ? semVer.lt(coercedUI5Version, changesPreviewToVersion) : false,
         changesLoader: feApp.service.version === OdataVersion.v2
     };
 
     // Add new files from templates e.g.
     const rootTemplatesPath = join(__dirname, '..', 'templates');
     // Add templates common to all template types
-    // Common files
-    const ignore = [feApp.appOptions?.typescript ? '**/*.js' : '**/*.ts'];
+    const jsIgnoreGlob = ['**/*.ts'];
+
+    let ignore = jsIgnoreGlob;
+    if (feApp.appOptions?.typescript === true) {
+        // isV2FETypesAvailable - Boolean to indicate if V2 Fiori Element types were available in the UI5 version
+        const isV2FETypesAvailable = feApp.ui5?.version ? semVer.gte(coercedUI5Version, V2_FE_TYPES_AVAILABLE) : false;
+        const tsIgnoreGlob = ['**/*.js'];
+        ignore = tsIgnoreGlob;
+        // Add local ui5.d.ts if types are missing in UI5 version for V2 Odata services
+        // OR template is OVP
+        if (feApp.service.version === OdataVersion.v2) {
+            if (isV2FETypesAvailable) {
```

Comment:
Should this be `!isV2FETypesAvailable` instead?
Also can the scope of `isV2FETypesAvailable` be reduced by moving closer to the usage?

Reply:
The code location has moved, hence the `Outdated` label.
I'm still wondering about this: https://github.com/SAP/open-ux-tools/pull/845#discussion_r1158804536